### PR TITLE
Add CephFS volume source to describe printer.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@rootfs PTAL .

**Comment by childsb** [comment]
@k8s-bot ok to test

**Comment by humblec** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by humblec** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by humblec** [comment]
@k8s-bot kops aws e2e test this

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -850,6 +852,17 @@ func printRBDVolumeSource(rbd *api.RBDVolumeSource, w PrefixWriter) {
 		rbd.CephMonitors, rbd.RBDImage, rbd.FSType, rbd.RBDPool, rbd.RadosUser, rbd.Keyring, rbd.SecretRef, rbd.ReadOnly)
 }
 
+func printCephFSVolumeSource(cephfs *api.CephFSVolumeSource, w PrefixWriter) {
+	w.Write(LEVEL_2, "Type:\tRBD (a Rados Block Device mount on the host that shares a pod's lifetime)\n"+
```
Comment (rootfs) [inline pkg/printers/internalversion/describe.go]: Cephfs

Comment (humblec) [inline pkg/printers/internalversion/describe.go]: Done :). Thanks

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -850,6 +852,17 @@ func printRBDVolumeSource(rbd *api.RBDVolumeSource, w PrefixWriter) {
 		rbd.CephMonitors, rbd.RBDImage, rbd.FSType, rbd.RBDPool, rbd.RadosUser, rbd.Keyring, rbd.SecretRef, rbd.ReadOnly)
 }
 
+func printCephFSVolumeSource(cephfs *api.CephFSVolumeSource, w PrefixWriter) {
+	w.Write(LEVEL_2, "Type:\tRBD (a CephFS mount on the host that shares a pod's lifetime)\n"+
```
Comment (rootfs) [inline pkg/printers/internalversion/describe.go]: still rbd :)

Comment (humblec) [inline pkg/printers/internalversion/describe.go]: Done. :)

**Comment by humblec** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

File: pkg/printers/internalversion/describe.go [hunk 3]
```diff
@@ -732,6 +732,8 @@ func describeVolumes(volumes []api.Volume, w PrefixWriter, space string) {
 			printPortworxVolumeSource(volume.VolumeSource.PortworxVolume, w)
 		case volume.VolumeSource.ScaleIO != nil:
 			printScaleIOVolumeSource(volume.VolumeSource.ScaleIO, w)
+		case volume.VolumeSource.CephFS != nil:
+			printCephFSVolumeSource(volume.VolumeSource.CephFS, w)
```
Comment (xiangpengzhao) [inline pkg/printers/internalversion/describe.go]: Also need to add `printCephFSVolumeSource` in `describePersistentVolume`. And I sent a PR #46210 to add other volume sources.

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: pkg/printers/internalversion/describe.go (hunk 3 above)
Comment (rootfs) [inline pkg/printers/internalversion/describe.go]: good catch @xiangpengzhao

Comment (humblec) [inline pkg/printers/internalversion/describe.go]: Thanks @xiangpengzhao . I added to pv describe as well.

**Comment by humblec** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by humblec** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by humblec** [comment]
@adohe Can you please review/approve this PR ?

**Comment by fabianofranz** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Review by droot — COMMENTED** [review]
LGTM

**Review by droot — APPROVED** [review]

**Comment by fabianofranz** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xiangpengzhao** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by xiangpengzhao** [comment]
The fix of unit test flake is #46383

**Pull request merged**